internetarchive/openlibrary
# Feature/observations vue component

Improve UI of the community book tags form (formally known as the observations form).  Original form has been replaced by a Vue component.  Further improvements to the component's style should be made before this is widely released.

### Technical

The new ObservationsForm.vue component is comprised of three main components (listed in order of appearance):

__Selections.vue__: Displays all of a patron's book tags for a work.  Tags are displayed as __Chip__ components that can be removed by clicking an 'X' symbol.  When a Chip is removed, an AJAX call that deletes the observation is made to our server.
__Categories.vue__: Displays all of the types of tags that can be chosen from.  Tag types are displayed in selectable __Chip__ components.  If a category Chip is clicked, all of its possible values are displayed in a __ValueCard__ component.
__ValueCard.vue__:  This is comprised of a __CardHeader__, which displays more detailed information about a type, and a __CardBody__, which displays all of the type's possible values as selectable Chips.  Some types allow multiple chips to be selected.  When a chip is selected or deselected, an AJAX request to update the patron's tags is made to our server.

This PR also adds a `build-assets:components` npm script, which runs `make components`.

### Testing

While logged in as a beta-tester:
1. Navigate to a book page and scroll to the "Reader Observations" component near the bottom of the page.
2. Click the link that prompts you to share your observations.
3. Ensure that any previously selected observations are present in the "Selections" section.
4. Select some new observations.  Ensure that the form updates correctly, and that the values are being persisted in the DB (may require a refresh).
5. Ensure that the components also work properly on the "My Observations" view. A link to this view can be found in the sidebar of the Reading Log page.

### Screenshot

![book_tags_modal](https://user-images.githubusercontent.com/28732543/128105089-fe442e94-376c-4a30-a64b-60d4e27939fe.png)

### Stakeholders
